src/corva/handlers.py

```diff
@@ -378,6 +395,136 @@ def wrapper(
+            app_stream_id=event.data.app_stream_id,
+            app_key=SETTINGS.APP_KEY,
+            app_connection_id=event.data.app_connection_id,
+        )
+        rerun_asset_cache_hash_name = get_cache_key(
+            provider=SETTINGS.PROVIDER,
+            asset_id=event.data.rerun_asset_id,
+            app_stream_id=event.data.rerun_app_stream_id,
+            app_key=SETTINGS.APP_KEY,
+            app_connection_id=event.data.rerun_app_connection_id,
+        )
+        internal_cache_hash_name = get_cache_key(
+            provider=SETTINGS.PROVIDER,
+            asset_id=event.data.rerun_asset_id,
+            app_stream_id=event.data.rerun_app_stream_id,
+            app_key=SETTINGS.APP_KEY,
+            app_connection_id=event.data.app_connection_id,
+        )
+
+        asset_cache = UserRedisSdk(
+            hash_name=asset_cache_hash_name,
+            redis_dsn=SETTINGS.CACHE_URL,
+            redis_client=redis_client,
+        )
+        rerun_asset_cache = UserRedisSdk(
+            hash_name=rerun_asset_cache_hash_name,
+            redis_dsn=SETTINGS.CACHE_URL,
+            redis_client=redis_client,
+        )
+        app_event = PartialRerunMergeEvent(
+            **event.data.dict(), event_type=event.event_type
+        )
 
+        with LoggingContext(
+            aws_request_id=aws_request_id,
+            asset_id=event.data.asset_id,
+            app_connection_id=event.data.app_connection_id,
+            handler=CorvaLoggerHandler(
+                max_message_size=SETTINGS.LOG_THRESHOLD_MESSAGE_SIZE,
+                max_message_count=SETTINGS.LOG_THRESHOLD_MESSAGE_COUNT,
+                logger=CORVA_LOGGER,
+                placeholder=" ...",
+            ),
+            user_handler=handler,
+            logger=CORVA_LOGGER,
+        ):
+            result = service.run_app(
+                has_secrets=event.has_secrets,
+                app_key=SETTINGS.APP_KEY,
+                api_sdk=CachingApiSdk(
+                    api_sdk=CorvaApiSdk(api_adapter=api),
+                    ttl=SETTINGS.SECRETS_CACHE_TTL,
+                ),
+                cache_sdk=InternalRedisSdk(
+                    hash_name=internal_cache_hash_name, redis_client=redis_client
+                ),
+                app=functools.partial(
+                    cast(
+                        Callable[
+                            [PartialRerunMergeEvent, Api, UserRedisSdk, UserRedisSdk],
+                            Any,
+                        ],
+                        func,
+                    ),
+                    app_event,
+                    api,
+                    asset_cache,
+                    rerun_asset_cache,
+                ),
+            )
+
+        return result
+
+    HANDLERS[RawPartialRerunMergeEvent] = wrapper
     return wrapper
+
+
+def _get_custom_event_type_by_raw_aws_event(
+    aws_event: Any,
+) -> Optional[Type[RawPartialRerunMergeEvent]]:
```

Comment:
can we make this function return event type and function that handles it to simplify logic in `base_handler`?

Reply:
We can. I've updated the function.